include/dlaf/eigensolver/bt_band_to_tridiag/impl.h

```diff
@@ -38,111 +43,158 @@ namespace dlaf::eigensolver::internal {
+  using namespace blas;
 
-  auto tfactor_task = [](const auto& tile_taus, const auto& tile_v, auto tile_t) {
-    using namespace lapack;
+  auto [tile_v_c, tile_t_c] = computeVT(b, tile_hh, std::move(tile_v), std::move(tile_t));
 
-    // taus have to be extracted from the compact form (i.e. first row of the input tile)
-    std::vector<T> taus;
-    taus.resize(to_sizet(tile_v.size().cols()));
-    for (SizeType i = 0; i < to_SizeType(taus.size()); ++i)
-      taus[to_sizet(i)] = tile_taus({0, i});
+  dlaf::tile::internal::trmm3(Side::Right, Uplo::Upper, Op::NoTrans, Diag::NonUnit, T(1), tile_t_c,
+                              tile_v_c, tile_w);
+  return std::make_tuple(std::move(tile_v_c), std::move(tile_w));
+}
 
-    const auto n = tile_v.size().rows();
-    const auto k = tile_v.size().cols();
-    larft(Direction::Forward, StoreV::Columnwise, n, k, tile_v.ptr(), tile_v.ld(), taus.data(),
-          tile_t.ptr(), tile_t.ld());
+template <Backend B, class T>
+struct single_tile_f;
 
-    return matrix::Tile<const T, Device::CPU>(std::move(tile_t));
-  };
-  return dlaf::internal::whenAllLift(ex::keep_future(std::move(tile_taus)),
-                                     ex::keep_future(std::move(tile_v)), std::move(mat_t)) |
-         dlaf::internal::transform(dlaf::internal::Policy<Backend::MC>(), tfactor_task) |
-         ex::make_future();
-}
+template <class T>
+struct single_tile_f<Backend::MC, T> {
+  void operator()(const matrix::Tile<const T, Device::CPU>& tile_v,
+                  const matrix::Tile<const T, Device::CPU>& tile_w, matrix::Tile<T, Device::CPU> tile_w2,
+                  matrix::Tile<T, Device::CPU> tile_e) {
+    using namespace blas;
+    using tile::internal::gemm;
+    // W2 = V* . E
+    gemm(Op::ConjTrans, Op::NoTrans, T(1), tile_v, tile_e, T(0), tile_w2);
+    // E -= W . V2
+    gemm(Op::NoTrans, Op::NoTrans, T(-1), tile_w, tile_w2, T(1), tile_e);
+  }
+};
 
-template <Backend backend, class TSender, class VSender>
-auto computeW(pika::threads::thread_priority priority, TSender&& tile_t, VSender&& tile_v) {
-  using namespace blas;
-  using T = dlaf::internal::SenderElementType<VSender>;
-  dlaf::internal::whenAllLift(Side::Right, Uplo::Upper, Op::NoTrans, Diag::NonUnit, T(1),
-                              std::forward<TSender>(tile_t), std::forward<VSender>(tile_v)) |
-      tile::trmm(dlaf::internal::Policy<backend>(priority)) |
-      pika::execution::experimental::start_detached();
-}
+template <class T>
+struct single_tile_f<Backend::GPU, T> {
+  void operator()(cublasHandle_t handle, const matrix::Tile<const T, Device::GPU>& tile_v,
+                  const matrix::Tile<const T, Device::GPU>& tile_w,
+                  matrix::Tile<T, Device::GPU>& tile_w2, matrix::Tile<T, Device::GPU>& tile_e) {
+    using namespace blas;
+    using tile::internal::gemm;
+    // W2 = V* . E
+    gemm(handle, Op::ConjTrans, Op::NoTrans, T(1), tile_v, tile_e, T(0), tile_w2);
+    // E -= W . V2
```

Comment:
```suggestion
    // E -= W . W2
```
? If yes, I think there's another comment with `V2` as well.

More importantly (and not for this PR), would it be useful to be able to express this as
```
gemm(Policy<B>(), ...) |
  gemm(Policy<B>(), ...);
```
, assuming it can be done equivalently (i.e. with the two gemms submitted immediately on the same stream)? The details are obviously in the `...`...

Reply:
Sorry for leaving this open. This comment does not block the PR.